pulsar-sql/presto-pulsar/src/main/java/org/apache/pulsar/sql/presto/decoder/protobufnative/PulsarProtobufNativeColumnDecoder.java

```diff
@@ -192,6 +194,12 @@ public long getLong() {
                 return floatToIntBits((Float) value);
             }
 
+            //return seconds field of protobuf/timestamp
+            if (columnType instanceof TimestampType && value instanceof DynamicMessage) {
+                DynamicMessage message = (DynamicMessage) value;
+                return (long) message.getField(message.getDescriptorForType().findFieldByName("seconds"));
```

Comment:
What is the difference between the following pattern:

```
            if (columnType instanceof TimestampType && value instanceof Timestamp) {
                return ((Timestamp) value).getSeconds();
            }
```

Reply:
I’ll check it, but I confirmed `value instanceof DynamicMessage` is true, and protobuf/timestamp not extends DynamicMessage, if I use `value instanceof DynamicMessage`, it seems not work

Reply:
I’ll check it, but I confirmed `value instanceof DynamicMessage` is true, and protobuf/timestamp not extends DynamicMessage, if I use `value instanceof Timestamp`, it seems not work

Reply:
I think we should return time in the unit of milliseconds here.  It seems that we need to convert it here. And do we also need to handle `Timestamp. nanos` here?